test/functional/test_framework/wallet.py

```diff
@@ -212,16 +212,20 @@ def get_address(self):
         assert_equal(self._mode, MiniWalletMode.ADDRESS_OP_TRUE)
         return self._address
 
-    def get_utxo(self, *, txid: str = '', vout: Optional[int] = None, mark_as_spent=True) -> dict:
+    def get_utxo(self, *, txid: str = '', vout: Optional[int] = None, mark_as_spent=True, avoid_immature_coinbase=True) -> dict:
         """
         Returns a utxo and marks it as spent (pops it from the internal list)
 
         Args:
         txid: get the first utxo we find from a specific transaction
         """
         self._utxos = sorted(self._utxos, key=lambda k: (k['value'], -k['height']))  # Put the largest utxo last
+        coinbase_filter = filter(lambda utxo: not utxo['coinbase'] or COINBASE_MATURITY <= utxo['confirmations'], self._utxos)
         if txid:
             utxo_filter: Any = filter(lambda utxo: txid == utxo['txid'], self._utxos)
+        elif len(list(coinbase_filter)) > 0 and avoid_immature_coinbase:
+            utxo_filter = filter(lambda utxo: not utxo['coinbase'] or COINBASE_MATURITY <= utxo['confirmations'], self._utxos)
+            utxo_filter = reversed(list(utxo_filter))
```

Comment:
Feels like there must be a way to express this without rewriting the filter? I think the only reason you need to do that here is because `list(coinbase_filter)` exhausts the iterator so you need to start it over. What if `coinbase_filter` was actually a list called `mature_coins` from the beginning of the function? I dunno, just an idea.